projects/vdk-core/src/vdk/internal/core/errors.py

```diff
@@ -607,32 +363,167 @@ def __error_type_to_actual_resolver(to_be_fixed_by: ResolvableBy) -> ResolvableB
     )  # What type is the error that caused this and whom to blame, Platform or Data Jobs Developer?
 
 
-def __wrap_exception_if_not_already(
-    to_be_fixed_by: ResolvableBy, msg: ErrorMessage, exception: BaseException
-):
-    if isinstance(exception, BaseVdkError):
-        # already wrapped
-        return exception
+# REPORTING
 
-    # TODO: how to assign cause (new_ex from old_ex) ?
-    if ResolvableBy.PLATFORM_ERROR == to_be_fixed_by:
-        return PlatformServiceError(msg)
-    elif ResolvableBy.USER_ERROR == to_be_fixed_by:
-        return UserCodeError(msg)
-    elif ResolvableBy.CONFIG_ERROR == to_be_fixed_by:
-        return VdkConfigurationError(msg)
-    else:
-        log.warning(
-            "Unknown to_be_fixed_by type. "
-            "This seems like a bug. We cannot wrap exception and return original one "
+
+def log_exception(log: Logger, exception: BaseException, *lines) -> None:
+    """
+    Log message and exception
+    """
+    message = "\n".join([s for s in lines if s])
+    log.warning(message)
+    log.exception(exception)
```

Comment:
Why are you logging it twice? 

log.exception would log the message at ERROR level (same as log.error) and print the stack race .
(reference: https://docs.python.org/3/library/logging.html#logging.Logger.exception)


`log.warning` is very much redundant in this case

Reply:
The idea is you pass a message to log at warn level, along with the exception. We use this three times in vdk-core, total, so we can re-visit it later when we audit all the log statements.